### PR TITLE
extensions\apidoc\helper\ApiIndexer.php - add Unicode support FileInfo(jsSearch) for bootstrap-guide mode.

### PR DESCRIPTION
**Pull request by keltstr**
extensions\apidoc\helper\ApiIndexer.php - add Unicode support FileInfo(jsSearch) for bootstrap-guide mode.

```
preg_match(/(.*)/s)
```

replaced 
    preg_match(/(._|\r_|\n*)/u)

### CODE REVIEW

File: extensions/apidoc/helpers/ApiIndexer.php [hunk 1]
```diff
@@ -25,18 +25,18 @@ class ApiIndexer extends Indexer
      */
     protected function generateFileInfo($file, $contents, $basePath, $baseUrl)
     {
-        // create file entry
-        if (preg_match('~<h1>(.*?)</h1>~s', $contents, $matches)) {
+		// create file entry
+        if (preg_match('~<h1>(.*|\r*|\n*?)</h1>~u', $contents, $matches)) {
```
Comment (samdark) [inline extensions/apidoc/helpers/ApiIndexer.php]: Shouldn't `.*` include `\r` and `\n`?

Comment (samdark) [inline extensions/apidoc/helpers/ApiIndexer.php]: If not, shouldn't `\R` be used instead?

Comment (cebe) [inline extensions/apidoc/helpers/ApiIndexer.php]: this has been handled by `/s` modifier before: http://php.net/manual/de/reference.pcre.pattern.modifiers.php
also adding `|` always needs new parenthesis: `(.*|\r*|\n*)`

**Comment by cebe** [comment]
What are you trying to solve with this change? .* should match any unicdeo character even without `u`...

**Comment by keltstr** [comment]
preg_match don't match utf8 without /u modifier (I was generating russian(utf8) guide with apidoc - and having empty fileInfo in jssearch index).
But /u and /s modifiers don't work together (/su or /us) - so I added /s modifier functionality using .|\r|\n and added /u modifier (tested ok)

**Comment by keltstr** [comment]
In the process of Guide generating is created file jssearch.index.js - it should contain `jssearch.files` variable, but after generating of Russian Guide(utf8) `jssearch.files` remains empty (jssearch.files="";). In English Guide `jssearch.files` normaly contains info about files and it used for output listing search result on a page.

**Comment by cebe** [comment]
I have reproduced the issue, if you enable PHP warnings in your php config which you definitively should, you get an exception about invalid UTF8.

**Pull request closed without merging**